editor/src/components/navigator/navigator.tsx

```diff
@@ -147,6 +147,10 @@ export const NavigatorComponent = React.memo(() => {
     }
   }, [selectionIndex, itemListRef])
 
+  React.useEffect(() => {
```

Comment:
Definitely worth adding a comment for this, as you'd have to click through to the code to even understand what `resetAfterIndex` is doing.